apps/duolingo/duolingo.star

```diff
@@ -0,0 +1,1580 @@
13TVjsDQdtRv7986p1ge3HT6pWh4mnH4RUrzCs7l+xbR99pmAAEQ6UQ5woWAB1aincJtqBzS45Sq6DzDyLFUdL7DE86TPzvBVi3eQGwA0FOTUb0AAAAAAAAAAAAEQAAEQAAMgAAAVZQOEzcAAAALxFABBD3oCiSpKbPIT/cRhd8MRJNIGrbRi6zEbjX8ccSRY0kRbU/dCAA7UhCxznzXwfAXQzNxmzbIAJFZ+Q3GJQAISgb6aYAsfGF9ZcNQ8UAsZEkRVLf3j02zjL77yc9WFAd0f8JoKM6AWDFzZxqZbR32Nu5lr6x+krnRzWNj2q4Udoqx6nKrlxJ2/TPV9929YVoN+n3l8+DX7lp/Z6h4kmAAQivWWFe27Xk0DmGXsPkrFaIcw0LoIDSyEK8T7AFWaFTC46iddD1JyLFUejjCm90mvjfK7DtywrgAEFOTUb0AAAAAAAAAAAAEQAAEQAAMgAAAVZQOEzcAAAALxFABBD3oCiSpKbPIT/cRhd8MRJNIGrbRi6zEbjX8ccSRY0kRbU/dCAA7UhCxznzXwfAXQzNxmzbIAJFZ+Q3GJQAISgb6aYAsfGF9ZcNQ8UAsZEkRVLf3j02zjL77yc9WFAd0f8JoKM6AWDFzZxqZbR32Nu5lr6x+krnRzWNj2q4Udoqx6nKrlxJ2/TPV9929YVoN+n3l8+DX7lp/Z6h4kmAAQivWWFe27Xk0DmGXsPkrFaIcw0LoIDSyEK8T7AFWaFTC46iddD1JyLFUejjCm90mvjfK7DtywrgAEFOTUb0AAAAAAAAAAAAEQAAEQAAMgAAAVZQOEzcAAAALxFABBD3oCiSpKbPIT/cRhd8MRJNIGrbRi6zEbjX8ccSRY0kRbU/dCAA7UhCxznzXwfAXQzNxmzbIAJFZ+Q3GJQAISgb6aYAsfGF9ZcNQ8UAsZEkRVLfHv43LvP67yc9WFAd0f8JoL06AWDFxRwrZTRX2Jupkq626kynWzEOt6K/UDZFDmORbXkmTd3dH13TVjsDQdtRv7986p1ge3HT6pWh4mnH4RUrzCs7l+xbR99pmAAEQ6UQ5woWAB1aincJtqBzS45Sq6DzDyLFUdL7DE86TPzvBVi3eQGwA0FOTUb0AAAAAAAAAAAAEQAAEQAAMgAAAVZQOEzcAAAALxFABBD3oCiSpKbPIT/cRhd8MRJNIGrbRi6zEbjX8ccSRY0kRbU/dCAA7UhCxznzXwfAXQzNxmzbIAJFZ+Q3GJQAISgb6aYAsfGF9ZcNQ8UAsZEkRVLfHv43LvP67yc9WFAd0f8JoL06AWDFxRwrZTRX2Jupkq626kynWzEOt6K/UDZFDmORbXkmTd3dH13TVjsDQdtRv7986p1ge3HT6pWh4mnH4RUrzCs7l+xbR99pmAAEQ6UQ5woWAB1aincJtqBzS45Sq6DzDyLFUdL7DE86TPzvBVi3eQGwA0FOTUb2AAAAAAAAAAAAEQAAEQAAMgAAAVZQOEzeAAAALxFABBD3oCiSpKbPIT/cRhd8MRJNIGrbRi6zEbjX8ccSRY0kRbU/dCAA7UhCxznzXwfAXQzNxmzbIAJFZ+Q3GJQAISgb6aYAsfGF9ZcNQwUOI9t2kgtmX35k+Np/nwRDBRH9nwCsxQCABAdlSqFsj5C1c3LfaO7JXNXTWNXDhgKItq7Hqa67AKAaAW6b/nTu2y4RsWLpJnk+bB6MIxQRbCp5vbrwCogItyRJtdRYgYN56MyGXjwYmynJRpnq2FVn63ujI1AnD0nH/h3ubBa47eGCzRv+u5Tyei+llLICQU5NRvYAAAAAAAAAAAARAAARAAAyAAABVlA4TN0AAAAvEUAEEOfAkJEkQTUip3O6x/AWT9MaBtlG6vurvMSrvMjYtJEkyZ0tjGW0/MOl8dz1/Adg1WcIf0eghvsbNTScnYiBEoEY3i2BGAQVCAPERpIUSX13j8vUMLPtv6MLDxZURfR/AuToKgB2l5s+pjO3O9RtSlvWyCufHs9xeLA/CQHK9uQwknMRIECKbevyelu2OaXwYD6P/kGderXCENI0PCvpdhCQrJq7Z2hGOcCKsZ+V/eK1mJymJ3UHospl0HQB9zsI7rV4Vrn+QqUpCz6v8C6nxP9ugO67bQAcAABBTk1G8AAAAAAAAAAAABEAABEAADIAAAFWUDhM1wAAAC8RQAQQ58CQkSRBNSKnc7rH8BZP0xoG2Ubq+6u8xKu8yNi0kSTJnS2MZbT8w6Xx3PX8B2DVZwh/R6CG+xs1NJydiIESgRjeLYEYBBUIA8RGkhRJfXePy9SwM+2/owsPFlRF9H8C5OgqAKrLTR/Tmdsd6jalLWvklU+P5zg82N/YtyeHkZz3K9vW5fW2bHMeAgLn6B/UqVdBHE3hWUg3tTgvmtUzNOPa2M/KfvESBggC6UmtQBRAToOmC1ijyHWwlt2zyPUXCk254/MK73JK/O8G6L7bBsABAEFOTUbwAAAAAAAAAAAAEQAAEQAAMgAAAVZQOEzXAAAALxFABBDnwJCRJEE1IqdzusfwFk/TGgbZRur7q7zEq7zI2LSRJMmdLYxltPzDpfHc9fwHYNVnCH9HoIb7GzU0nJ2IgRKBGN4tgRgEFQgDxEaSFEl9d4/L1LAz7b+jCw8WVEX0fwLk6CoAqstNH9OZ2x3qNqUta+SVT4/nODzY39i3J4eRnPcr29bl9bZscx4CAufoH9SpV0EcTeFZSDe1OC+a1TM049rYz8p+8RIGCALpSa1AFEBOg6YLWKPIdbCW3bPI9RcKTbnj8wrvckr87wbovtsGwAEAQU5NRvAAAAAAAAAAAAARAAARAAAyAAABVlA4TNcAAAAvEUAEEOfAkJEkQTUip3O6x/AWT9MaBtlG6vurvMSrvMjYtJEkyZ0tjGW0/MOl8dz1/Adg1WcIf0eghvsbNTScnYiBEoEY3i2BGAQVCAPE2nbbNnBP1BsKQey/qErKBB8R/Z8A2isTgKp0UcfQjO1K8jaFLKvFmU63+zjcsr9Qt
```

Comment:
Since the `hide_duolingo_in_rotation` variable is declared inside of an `if` statement, it ceases to exist after the`if` statement. Instead it should be declared outside of the statement:

```suggestion
        hide_duolingo_in_rotation = False
        if week_xp_scores_total == 0:
            hide_duolingo_in_rotation = True
            if display_view == "twoweeks":
                print("IMPORTANT: No Duolingo lessons have been completed in the last 14 days - Tidbyt display will be hidden.")
            else:
                print("IMPORTANT: No Duolingo lessons have been completed in the last 7 days - Tidbyt display will be hidden.")
```

Reply:
Thanks. Let me have a look. I only just added this feature since and admit I have not been able to test it properly. Let me see if I can find a Duolingo user who has not practiced recently to test. This feature is meant to hide the app if no lessons have been completed recently (ie within the last week). If the two week chart is being displayed it should be hidden if the user hasn't practiced for two weeks instead.

Reply:
Ok, I pushed a fix.